### ISSUE TITLE
Cloudformation Spec vs Documentation

### ISSUE THREAD

**Issue by kddejong** [awslabs/cfn-python-lint#44]
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-cloudfront-distribution-distributionconfig.html#cfn-cloudfront-distribution-distributionconfig-defaultcachebehavior

CloudFormation Spec has this value as not required but the documentation has it as required.  Which one is it and where does the fix go?

**Comment by cmmeyer** [comment]
I've alerted the service team of the issues with the resource spec.  Reclassified as a big and will leave open until the internal ticket is closed.

**Comment by farski** [comment]
This may be a similar issue (I'm not sure how to check the spec):

I have a custom resource

```
AppServiceDesiredCountRolloverResource:
    Type: "Custom::EcsServiceSizeRollover"
    Version: "1.0"
    Properties:
      ClusterName: !Ref ECSCluster
      .....
```

Which gets `Invalid resource attribute Version for resource AppServiceDesiredCountRolloverResource`

The [docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-cfn-customresource.html) for custom resources list a `"Version": "1.0"` top-level attribute on custom resource definitions.

**Comment by farski** [comment]
This one is perhaps a different case. For Lambda functions, the [Cfn docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-lambda-function.html#cfn-lambda-function-memorysize) **and** spec say the memory limit is `1536`, but the [Lambda docs](https://docs.aws.amazon.com/lambda/latest/dg/limits.html) and reality say the limit is `3008` (CloudFormation accepts templates with a value of `3008`).

**Comment by kddejong** [comment]
@farski the Version is on us.  I will work to fix that.  Sorry about that.  

What error are you getting for the Lambda one?  We should have updated our check to 3008.
https://github.com/awslabs/cfn-python-lint/blob/master/src/cfnlint/rules/resources/lmbd/FunctionMemorySize.py#L41  Accidentally used the CloudFormation docs the first time as you had mentioned but it should be fixed now.

but the other qualifier is it be a multiple of 64 which we also test for.

**Comment by farski** [comment]
@kddejong `You must specify a value that is greater than or equal to 128, and it must be a multiple of 64. You cannot specify a size larger than 1536. The default value is 128 MB at 3008`

(for ` MemorySize: 3008`)

**Comment by cmmeyer** [comment]
Yup, that's definitely an old version of the linter.  You can find that exact message here, with the higher max value, in the code:
https://github.com/awslabs/cfn-python-lint/blob/779ccc8c2430dc9091012255aacd2d74af873380/src/cfnlint/rules/resources/lmbd/FunctionMemorySize.py#L28

You may need to do a `pip install cfn-lint --upgrade` and make sure you're running v.0.2.0 or 0.2.1.

**Comment by kddejong** [comment]
@farski https://github.com/awsdocs/aws-cloudformation-user-guide/pull/68/ also I have submitted and now need to rebase that pull request.

**Comment by cmmeyer** [comment]
Ok, just merged the "version" piece.  You can build from code to get both fixes now, or wait until the next version to upgrade via pip.

**Comment by farski** [comment]
@cmmeyer Hm, I'll dig into this some more because when I upgrade I get `Requirement already up-to-date: cfn-lint in /usr/local/lib/python3.6/site-packages (0.2.1)`. I only first installed this two days ago.

**Comment by cmmeyer** [comment]
This is interesting -- that says python3.6, but your error on the VS code side was 2.7.  I wonder if there are some pip shenanigans going on here?

I noticed pip3 was having trouble upgrading me from 0.1.0 to 0.2.1 -- I did a bunch of `pip3 uninstall cfn-lint` and `pip3 install cfn-lint` without getting a new version.  

I would try to clean up all current versions on your machine (`pip2 uninstall cfn-lint` and/or `pip3 uninstall cfn-lint`) then `pip3 install cfn-lint --no-cache --upgrade` and see if you get a nice, clean install with the higher memory cap.
